orchestrator/careplanservice/handle_createtask.go

```diff
@@ -190,7 +190,21 @@ func (s *Service) handleCreateTask(ctx context.Context, request FHIRHandlerReque
 	return func(txResult *fhir.Bundle) (*fhir.BundleEntry, []any, error) {
 		var createdTask fhir.Task
 		result, err := coolfhir.FetchBundleEntry(s.fhirClient, txResult, func(_ int, entry fhir.BundleEntry) bool {
-			return entry.Response.Location != nil && strings.HasPrefix(*entry.Response.Location, "Task/")
+			if entry.Response.Location == nil {
+				return false
+			}
+			// HAPI uses relative Location URLs, Microsoft Azure FHIR uses absolute URLs.
+			createdResourceLocation := *entry.Response.Location
+			createdResourceLocation = strings.TrimPrefix(createdResourceLocation, s.fhirURL.String())
+			// depending on the base URL ending with slash or not, we might end up with a leading slash.
+			// Trim it for deterministic comparison.
+			createdResourceLocation = strings.TrimPrefix(createdResourceLocation, "/")
```

Comment:
Why not handle this in one trim with `s.fhirURL.String()+"/"` ?

Reply:
Say some FHIR service would return a local reference like `/Task/2`, in that case it would not strip the leading `/` by stripping the base url of the service including the `/`

Reply:
Ahhhhh yes you're right, fair point